Form Factors Explained first stab

anssiko opened the pull request:
<!--
no preview
-->

[Preview](https://raw.githack.com/webscreens/form-factors/first-stab/index.html)

anssiko committed:
Form Factors Explained first stab

kenchris commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,21 @@
+# Second Screen Community Group
+
+This repository is being used for work in the W3C Second Screen Community Group, governed by the [W3C Community License Agreement (CLA)](https://www.w3.org/community/about/agreements/cla/). To make substantive contributions, you must join the CG. 
```
join -> first join?

anssiko replied:
This is a W3C template, so I wouldn't edit it.

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+<section id="abstract">
+## Abstract
+
+This explainer discusses new device [=form factors=] that are emerging and related terminology that is being established. This document proposes one naming scheme to be able to reason about and use these terms and concepts consistently in [related web specifications](#related-web-specifications). The use of these terms in other contexts may differ.
```
not wrapping these lines makes it really hard to review on github

anssiko replied:
I propose use [Preview](https://raw.githack.com/webscreens/form-factors/first-stab/index.html).

Tidy does not work with markdown, sadly. I'm open to PRs if someone wants to manually wrap the lines :)

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+## Terminology and conceptual model
+
+><dfn data-export="">Form factor</dfn> is a hardware design aspect that defines and prescribes the size, shape, and other physical specifications of components, particularly in electronics. A form factor may represent a broad class of similarly sized components, or it may prescribe a specific standard. It may also define an entire system, as in a computer form factor. ([Wikipedia](https://en.wiki ...
```
shouldn't start with >

anssiko replied:
It is a comment.

>Like this :)

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+A [=foldable device=] has a mechanism that allows the device to fold open and closed similarly to traditional laptops. To differentiate from the traditional laptop form factor, a foldable device has a screen on both sides of the fold. In the most typical configuration, a foldable device can fold a one bigger screen into two smaller screens. A foldable device can have one or more physical internal ...
+
+<img src="images/horizontal-fold.svg" height="150px" alt="">
```
should probably add alt text "Horizontal fold" etc

anssiko replied:
Fixed.

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+A <dfn>posture</dfn> is defined as the primary way in which the [=foldable device=] is positioned and presented to the user. Because [=foldable devices=] have a hinge they can present different [=postures=] depending on the hinge angle, device orientation, and the way on how the device is being held by the user. Example postures include “book”, “flat”, “laptop”, and “tent”. The list of postures i ...
+
+![](images/postures.png)
+*<br>Example postures from left to right: “closed”, “laptop”, “table top”, “flat”, “canvas”, and “book”. ([Source](TODO))*
```
We should probably use the same nomenclature... like screen fold spec uses tablet for canvas, but maybe canvas is better... I have also seen "compose" being used instead of laptop (which doesn't make sense for a phone)

anssiko replied:
Can you share a concrete proposal how to word this? Or should this be an issue instead?

diekus replied:
+1 here. We've discussed the need to rethink the name of the postures to make them as abstract as possible. The fact that 'laptop' is both an actual device and a 'posture' mode can be confusing and misleading. Can a laptop have another posture that's not laptop? Does a mobile device get into laptop posture? Similar case with tablet as a device type and posture. Happy to help Kenneth abstract this nomenclature to broaden it.

anssiko replied:
I made @diekus's comment an issue in the spec in the section that defines posture to make sure this does not go unnoticed.

(Note: images/postures.png was removed in 3229fe0eda3c595795057ca5c1267e8962f03c71)

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+### Spanning
+
+<dfn>Spanning</dfn> is a concept that allows apps or websites to run side by side on a [=dual-screen device=]. The ability to maximize a given application across the two [=screens=] is enabled through the [=window manager=]. Because spanning across the two screens double the screen estate it is expected that applications and websites leverage that additional space to optimize or enhance the UX to ...
```
Do window segments before... maybe rename to screen segments.. then you can say that spanning spans across these segments

anssiko replied:
Moved sections and added "spanning web content across [=window segments=]"

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+### Window segments
+
+A <dfn>window segment</dfn> represents the regions (and their dimensions) of the window that reside on separate (adjacent) [=displays=].
```
With something like FancyZones this doesn't necessarily need to be separate physically displays

arskama replied:
Can we have a picture example?
display or screen? which term to use?

anssiko replied:
OK.

anssiko replied:
@arskama's feedback suggests we need to improve the definitions of "display" and "screen". These terms are overloaded and used differently in different contexts in general (e.g. software vs. hardware, formal vs. informal use). I believe we should define these terms in a way they make sense to use in the context of the specs in scope of this explainer.

@arskama, can you investigate this? Maybe we this could be an issue of its own. Feel free to open one.

arskama replied:
@anssiko ok, I ll suggest something in context of these specs.

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+### Viewport
+
+<div class="issue">
+Ensure [Visual Viewport API](#visual-viewport-api) defines the two conceptual viewports, colloquially known as the [=visual viewport=] and [=layout viewport=], in a manner compatible with the [related web specifications](#related-web-specifications).
```
Ensure? weird start

anssiko replied:
Reworded.

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+One CSS pixel doesn’t necessarily correspond to one physical screen pixel, since that would make everything very small on high resolution screens. Instead one such CSS pixel might correspond to for example 2x2 square of physical pixels - or even non-discrete values given a subpixel (red, green, and blue) layout.
+
+Many OS’s allow to change the scaling of all UI elements which often is reflected in the “device pixel ratio” that makes up one CSS pixel. A browser also often allows the user to apply a scale on top of the OS-provided scale on a per site basis. The layout is laid out considering the browser viewport in CSS pixels. For example, given a browser scale of 125%, a 800px wide viewport becomes a 640px  ...
```
link to devicepixelratio on mdn?

anssiko replied:
Added https://drafts.csswg.org/cssom-view/#dom-window-devicepixelratio

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+### Content scaling
+
+Usually content is laid out given the size of the viewport in [=CSS pixels=], but that may not always be the case. Especially on mobile devices the user may select “Show desktop site” which will usually lay the content out mimicking common desktop browser window size (typically 980-1024px width), and then that content is scaled so that the width fits within the actual viewport.
```
this is what you call zoom layout, which we could call layout scaling or so

anssiko replied:
Are you suggesting we rename this section, or define a new term?

kenchris commented on index.html:
```diff
@@ -0,0 +1,368 @@
+Zouhir Chahoud,
+Kenneth Rohde Christiansen,
+Victor Costan,
+Francois Daoust,
```
Isnt Francois with ç

anssiko replied:
Fixed :)

kenchris commented:
I did a raw review, but can you please wrap lines so that I can actually read it on Github :-)

arskama commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+*<br>A concept of a [=foldable-screen device=] on the left and a concept of a [=dual-screen device=] on the right. ([Source](https://w3c.github.io/screen-fold/))*
+
+A [=foldable device=] has a mechanism that allows the device to fold open and closed similarly to traditional laptops. To differentiate from the traditional laptop form factor, a foldable device has a screen on both sides of the fold. In the most typical configuration, a foldable device can fold a one bigger screen into two smaller screens. A foldable device can have one or more physical internal ...
```
" Screens made of flexible material can also bend within certain limits and to differentiate from the designs with a physical hinge, they do not have a physical gap between the logical regions divided by the fold."

This sentence is confusing.
The wording "to differentiate" doesn't seem to be proper.
single flexible material screen, do also have a hinge, not visible, but they do have.
Therefore I would suggest that:
- All foldable devices have a physical hinge.
- Multi/Dual screens devices have a VISIBLE physical hinge with a gap
- Foldable screen devices have a NON VISIBLE physical hinge with no gap.
Maybe redundant wording here, if we consider that  "visible hinge" = "gap."

anssiko replied:
Good suggestions. I reworded this section, PTAL. I have a feeling it's be nice to have a complementary table with form factors (dual-screen device, foldable-screen device) on one axis and features (visible physical hinge, gap etc.) on the other. Maybe something to add in subsequent updates.

arskama commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+### Dual-screen device
+
+A <dfn data-export="">dual-screen device</dfn> has two or more physical internal screens made of solid materials placed together to form a larger foldable virtual screen. A defining characteristic of this form factor is a physical hinge that can rotate from 0 degree to at most 360 degrees that forms a physical gap in between the screens. The range of the hinge movement as well as the width of the ...
```
Can we change it to multiscreen, or have there been already a discussion on that?
I don't think that 2 or 3 screens would change the content of these specs?
I mean calling dual-screen and already mentioning on the first sentence that we can have 2 or more screens is confusing.

anssiko replied:
That is a good catch. I added:
`A [=dual-screen device=] is an instance of a <dfn data-export="">multi-screen device</dfn> with two screens.`

This topic would be good to be discussed separately i.e. which specs can scale to N screens, and which hardcode assumptions of two screens. Thinking of future-proofing, "two should be enough for everyone" will probably not stand the test of time :)

arskama commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+### Foldable-screen device
+
+A <dfn data-export="">foldable-screen device</dfn> has one or multiple screens made of foldable material that can bend so that the screen can fold onto itself. The hinge can typically rotate from 0 degree to 180 degrees depending on the flexible screen materials’ capability of sustaining the bend radius. The material around the area of the screen that can bend is often softer than on the other pa ...
```
- statement 1:
 "A foldable-screen device has one or multiple screens made of foldable material that can bend so that the screen can fold onto itself."

- Statement 2:
"To differentiate from a dual-screen device, a foldable-screen device does not have a physical gap between the screens that form the logical regions."

How does this work? do you have an example including statement 1 (multiple screens) and 2 ?

anssiko replied:
Are you concerned about the "or multiple" bit in 1? I clarified that by "the each screen can". There could be a device with two or more foldable screens in one device.

Re 2, can you specify the unclear bit?

arskama replied:
sorry for the confusion.
In this foldable-screen device description, there are, at the moment 2 statements defining how they should look like to be qualified as foldable-screen device:
1) "A foldable-screen device has one or multiple screens made of foldable material that can bend so that the screen can fold onto itself."
That s fine and that s also valid as a generic statement for all generic foldable devices.

but 2) "To differentiate from a dual-screen device, a foldable-screen device does not have a physical gap between the screens that form the logical regions."

That s also fine when we are talking about a single "flexible" screen, having a flex material in the middle such as "Samsung galaxy fold".

But what kind of device would be a foldable-screen device, that has more than 1 screen, with no physical gap? (complying with both statements above)? Do you have an existing example?

anssiko replied:
Thanks for paying attention to the details. Addressed by 9dbff3e302507ed7a7b1ce2b124cd61303622b83.

arskama commented on index.html:
```diff
@@ -0,0 +1,368 @@
+*<br>A [=foldable-screen device=] as a laptop. ([Source](TODO))*
+
+![](images/foldable-screen-device-mobile.png)
+*<br>Another [=foldable-screen device=] as a mobile phone. ([Source](TODO))*
```
what makes this device foldable-screen and not dual-screen?

anssiko replied:
No gap between the screens.

arskama replied:
Not very visible from the picture.

arskama commented on index.html:
```diff
@@ -0,0 +1,368 @@
+
+### Screen
+
+<dfn>Screen</dfn> is a synonym to [=display=]. [=Screen=] is used more in software context (logical) while display is primarily used in hardware context (physical). See also the [=Screen interface=].
```
How do we differentiate with:
Screen - the aggregate 2D space occupied by all the connected displays.
From https://github.com/webscreens/window-segments/blob/master/EXPLAINER.md

anssiko replied:
We don't currently, but as part of this exercise need to. The tool flags the duplicate definitions, see the red pill at the top when in HTML Preview.

michaelwasserman replied:
Yeah, we need a term for the "overall visual workspace" of the device, whatever we choose to call the composition of multiple physical displays/screens.

Wikipedia says ["Multi-monitor, also called multi-display and multi-head, is the use of multiple physical display devices, such as monitors, televisions, and projectors, in order to increase the area available for computer programs running on a single computer system."](https://en.wikipedia.org/wiki/Multi-monitor)
CSSOM View's ["web-exposed-screen-information"](https://drafts.csswg.org/cssom-view/#web-exposed-screen-information) only uses singular output device terminology
MDN says ["The Screen interface represents a screen, usually the one on which the current window is being rendered... browsers determine which screen to report as current by detecting which screen has the center of the browser window."](https://developer.mozilla.org/en-US/docs/Web/API/Screen)
MSDN says ["The bounding rectangle of all the monitors is the _virtual screen_."](https://docs.microsoft.com/en-us/windows/win32/gdi/the-virtual-screen)
Apple says ["Maximize your workspace with extended desktop mode, which lets you enjoy full-screen apps and windows on each monitor"](https://support.apple.com/en-us/HT202351)
Ubuntu says ["With two screens, these display modes are available: Join Displays: screen edges are joined so things can pass from one display to another..."](https://help.ubuntu.com/stable/ubuntu-help/display-dual-monitors.html.en)
Arch Linux says ["Display refers to a collection of screens" and more...](https://wiki.archlinux.org/index.php/Multihead)
Chromebook Help says ["You can connect your Chromebook to a monitor or TV"](https://support.google.com/chromebook/answer/1060909?hl=en) :-/
Android mentions ["Multi-Display"](https://source.android.com/devices/tech/display/multi_display) and ["multi-screen"](https://source.android.com/devices/tech/display/multi_display/recommended-practices#foldable)
iOS/UIKit says ["The user can connect additional screens"](https://developer.apple.com/documentation/uikit/windows_and_screens/displaying_content_on_a_connected_screen)
Chromium code says ["A single Screen object exists regardless of the number of connected displays."](https://source.chromium.org/chromium/chromium/src/+/master:ui/display/screen.h;l=36)

arskama replied:
I was about to comment that Mike s comment on spanning using the word: "visual workspace" sounded good.

arskama replied:
"screen/visual workspace: the aggregate 2D space composed by all the active connected screens"
(some native English speaker might find more accurate wording)

anssiko replied:
Added a placeholder for the "overall visual workspace" term (along with the very helpful references) in acda45b.

anssiko committed:
Address review feedback

anssiko committed:
Address more review feedback

anssiko commented:
Thanks for your review @kenchris and @arskama! Your feedback has been addressed in 9c83f19 and a8d9a00 respectively.

I'm aware of the pain reviewing this PR without lines wrapped. https://github.com/htacg/tidy-html5 does not understand this custom markdown in HTML syntax we're experimenting with for this doc, so we need to investigate other alternative for our tidying purposes.

anssiko committed:
Exclude non-Creative Commons images for now

We can add the images later with a permission from copyright owners.

anssiko commented:
Reviewed-by: @kenchris @arskama 

PTAL: @michaelwasserman @zouhir @dlibby- @diekus @pwnall @darktears @lauramorinigo

This first stab does not need to be perfect, so looking for high-level review comments whether we're going to the right direction with this. We'll iterate on the specifics in the subsequent updates.

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,353 @@
+
+><dfn data-export="">Form factor</dfn> is a hardware design aspect that defines and prescribes the size, shape, and other physical specifications of components, particularly in electronics. A form factor may represent a broad class of similarly sized components, or it may prescribe a specific standard. It may also define an entire system, as in a computer form factor. ([Wikipedia](https://en.wiki ...
+
+![](images/relationship.svg)
```
Perhaps "This explainer" should be "Form factors", "Screen form factors" or "Form factors considered in this explainer" or similar? Should we capture single-screen form factors here (or more explicitly in the explainer)?

diekus replied:
interesting thought. can you elaborate a bit more?

michaelwasserman replied:
1) It might help to clarify the "This explainer" node.
2) Defining terminology for a "traditional single-screen device" form factor (or similar) could help us more clearly convey the basis for existing paradigms (like window.screen) and how new form factors differ from that base state. Perhaps this definition could refer to the CSSOM View Draft's ["2.3. Web-exposed screen information"](https://drafts.csswg.org/cssom-view/#web-exposed-screen-information), which describes "the screen of the output device". MDN also [says](https://developer.mozilla.org/en-US/docs/Web/API/Screen) "The Screen interface represents a screen, usually the one on which the current window is being rendered, and is obtained using window.screen... Note that browsers determine which screen to report as current by detecting which screen has the center of the browser window." In traditional computing environments, this form factor was obvious or implicit (a desktop's single monitor, a laptop's single internal display panel, a phone or tablet's single screen). Making explicit terminology for that traditional form factor could help us more clearly define the distinguishing characteristics of multi-monitor setups, foldable devices, and new OS/WM composite/virtual screen paradigms.

anssiko replied:
Addressed by 643691c.

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,353 @@
+
+A <dfn data-export="">foldable-screen device</dfn> has one or multiple screens made of foldable material that can bend so that the each screen can fold onto itself. The hinge can typically rotate from 0 degree to 180 degrees depending on the flexible screen materials’ capability of sustaining the bend radius. The material around the area of the screen that can bend is often softer than on the oth ...
+
+### Multi-monitor setups
```
I wonder if we should use the term "multi-screen device", "extended-screen device", or similar instead. Fine to leave it as-is for now...

diekus replied:
Not sure if multi-screen works here since this category is specifically for devices that fold, and while some foldables can be multiscreen not all multiscreen are foldables.

anssiko replied:
"Multi-monitor setup" was derived from https://en.wikipedia.org/wiki/Multi-monitor If there is a better established term we should switch to that. It sounds like we should keep the "multi-monitor setup" for now.

I noted "extended-screen device" as a synonym, but should settle on a canonical term for the definition.

michaelwasserman replied:
Sounds good, thanks!

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,353 @@
+
+### Multi-monitor setups
+
+A <dfn data-export="">multi-monitor setup</dfn> is a traditional mainstream setup in which a device (foldable or not) has one or more monitors connected to it (wired or not). Major desktop operating systems provide affordances to the user to arrange the logical position of the monitors as seen by the OS to match the physical location of the monitors as selected by the user. In this configuration, ...
```
I attempted to rewrite this for clarity (and included a potential re-naming of the term), perhaps some of this can be broken out to describe a more generalized multi-display arrangement term?:

> An <dfn data-export="">extended-screen device</dfn> is any device with one or more physical displays connected to expand the default visual workspace. This includes traditional desktop, laptop, and mobile devices, with additional displays connected beyond the single screen typical of these form factors. Additional displays are often monitors, televisions, or projectors connected by a wire, but also includes other display form factors and connection technologies.

> Major desktop operating systems allow users to customize the logical arrangement of multiple screens, including their relative arrangement and other attributes. The arrangement may or may not reflect the placement and dimensions of underlying physical hardware. In this configuration, it is typical to mix and match monitors of different physical sizes and resolutions, in both landscape and portrait configurations, to optimize the screen space for the usage model. This free-form composition of screens is distinct from the nature of dual-screen devices, which have well-defined relationships between physical displays, and from folding-screens, which expands the operational modes of a single physical display.


Perhaps this content is more fitting as a note in in the window segments definition?

> Power users may also make use of advanced software tools to create virtual zones to which application windows can snap into (see e.g. [Microsoft PowerToys FancyZones](https://github.com/microsoft/PowerToys/wiki/FancyZones-Overview)). The virtual zones provide a similar abstraction to the [=window segments=].

anssiko replied:
Replaced the section with this rewrite proposed by @michaelwasserman with editorial changes.

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,353 @@
+
+### Spanning
+
+<dfn>Spanning</dfn> is a concept that allows apps or websites to run side by side on a [=dual-screen device=], spanning web content across [=window segments=]. The ability to maximize a given application across the two [=screens=] is enabled through the [=window manager=]. Because spanning across the two screens double the screen estate it is expected that applications and websites leverage that  ...
```
The first sentence isn't quite right, perhaps this paragraph needs some reworking. Foldables allow apps to run "side by side", but spanning itself is not that.

<dfn>Spanning</dfn> is a window mode on a [=dual-screen device=], where a single application or web content window is spread across both screens of the device. The ability to span a given application across the two [=screens=] is enabled through the [=window manager=]. Applications and websites may wish to optimize their layout, incorporating the physical gap of the [=dual-screen device=] hinge, to provide a better experience.

diekus replied:
+1ish . I agree that spanning in a PC-like scenario generally means across displays, but I think we need to be careful to allow 'spanning' to mean _across the border_ (whether this border is a display edge or a hinge or a screen). 

 The Screen Fold API can create beautiful innovative layouts with the help of the CSS spanning features on a Galaxy Fold for example, even though it doesn't have 2 screens, and the main concept here is that the developer can get some content to span or dock it alongside the hinge. in this case the spanning is across the hinge.

Spanning is more powerful if abstracted to a border than to a screen?

what are we spanning across, might be the question. Are we getting too philosophical here?  

to span or not to span... that is the question.

anssiko replied:
I adopted the @michaelwasserman's proposal for the first sentence and the note proposed by @diekus.

diekus commented:
It might be useful to add a note that states that in some cases a foldable device can also be a multi-monitor one. The first image on the conceptual model seems to indicate otherwise. There are already devices in the market that comply with this definition (Samsung Galaxy Fold and Fold 2) and I wouldn't discard seeing more devices. 

**"A device can be a foldable device and have a multi-monitor setup. These form factors are not mutually exclusive and depend on the hardware present on the device _and_ its use across software"**

I don't really think it'd be the case unless both (or more) screens are actively being used at the same time though, which interestingly is the case with some apps already. (selfie app and camera use internal and external displays on a galaxy fold & fold 2) and I'd assume it's a trend that we'll be seeing more.)

Also it might spark the interest on getting this feature to a web API to enable similar experiences on the web platform. <3

anssiko committed:
Address review feedback

anssiko committed:
Add 'overall visual workspace' term placeholder

anssiko committed:
Address more review feedback

anssiko commented:
Addressed review feedback from @michaelwasserman and @diekus in be0a6f6 and acda45b. Also addressed further feedback from @arskama in 9dbff3e.

Current status:

Reviewed-by: @kenchris @arskama @michaelwasserman @diekus @darktears

PTAL:  @zouhir @dlibby- @pwnall @lauramorinigo

You may find the [Preview](https://raw.githack.com/webscreens/form-factors/first-stab/index.html) easier to review than the HTML without line wrapping (to be fixed once we finish this first stab iteration).

Please send your feedback by 18 Nov 2020.

darktears commented:
LGTM.

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,386 @@
+- [Apple](https://support.apple.com/en-us/HT202351) says "Maximize your workspace with extended desktop mode, which lets you enjoy full-screen apps and windows on each monitor"
+- [Ubuntu](https://help.ubuntu.com/stable/ubuntu-help/display-dual-monitors.html.en) says "With two screens, these display modes are available: Join Displays: screen edges are joined so things can pass from one display to another..."
+- [Arch Linux](https://wiki.archlinux.org/index.php/Multihead) says "Display refers to a collection of screens" and more...
+- [Chromebook Help](https://support.google.com/chromebook/answer/1060909?hl=en) says "You can connect your Chromebook to a monitor or TV" :-/
```
Haha, please remove my ":-/" commentary :-)

anssiko replied:
Whoops, my copypasta error :-)

Fixed by 55c430a.

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,386 @@
+
+<dfn>Spanning</dfn> is a window mode on a [=dual-screen device=], where a single application or web content window is spread across both [=screens=] of the device. The ability to maximize a given application across the two [=screens=] is enabled through the [=window manager=]. Because spanning across the two screens double the screen estate it is expected that applications and websites leverage t ...
+
+<div class="note">
```
Perhaps this note belongs with one of the form factor definitions instead of with Spanning?

anssiko replied:
Fixed by a984764.

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,386 @@
+
+### Fold
+
+<dfn>Fold</dfn> is used as both a verb and a noun in the context of this explainer. _To fold_ means to lay one [part of the] screen over another [part of the] screen. _A fold_ (aka a screen fold) as a noun refers to the flexible [=screen area=] that bends.
```
Perhaps "refers to the flexible [=screen area=] that bends, or the hinge between the screens of a [=dual-screen device=]", if that's also a fold?

anssiko replied:
Addressed by 623fa72 with some tweaks. @kenchris reviewed and was happy.

michaelwasserman commented on index.html:
```diff
@@ -0,0 +1,386 @@
+- Access the hinge size, type and angle
+
+For [=multi-monitor setups=]:
+- Ability to layout and display a window on any of the connected screens
```
This first and third points seem like duplicates; I might change the points to:
For [=multi-monitor setups=], let web applications:
- Request information about all screens in the [=overall visual workspace=]
- Request fullscreen on any screen in the [=overall visual workspace=]
- Request placement of non-fullscreen windows on any screen in the [=overall visual workspace=]

anssiko replied:
Addressed by 13de491.

michaelwasserman reviewed (approved):
lgtm with a few more non-blocking comments.
Thank you, Anssi and everyone else for working on this!

michaelwasserman commented on README.md:
```diff
@@ -1,2 +1 @@
-# form-factors
-Foldables and Multi-Screen Explainer
+# Form Factors Explained
```
Can you add a link to the preview here?

anssiko replied:
Addressed by c31ffef.

(Once this PR is merged, it'll appear at https://webscreens.github.io/form-factors/)

anssiko committed:
Update Abstract, add 'traditional single-screen device' dfn placeholder

Link to 'traditional single-screen device' and 'traditional laptop'

anssiko committed:
Drop emoticon :-)

anssiko committed:
Move note to a more appropriate section

anssiko committed:
Clarify 'Fold' definition

anssiko committed:
Update multi-monitor setup bullets in 'What...?' section

anssiko committed:
Update README

anssiko committed:
Update CSS to improve doc readability

The stylesheet regressed probably due to this change:
https://github.com/w3c/tr-design/pull/224

Deploying this workaround to be lifted when the upstream fixes this.

anssiko commented:
Here's the announcement: https://lists.w3.org/Archives/Public/public-webscreens/2020Nov/0001.html 🎉
